src/Gui/Action.cpp

```diff
@@ -539,6 +540,24 @@ void WorkbenchGroup::addTo(QWidget *w)
         connect(_group, SIGNAL(triggered(QAction*)), box, SLOT(onActivated (QAction*)));
         bar->addWidget(box);
     }
+    else if (w->inherits("QMenuBar")) {
+        auto menuBar = qobject_cast<QMenuBar*>(w);
+        ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/MainWindow");
+        
+        QComboBox* box = new WorkbenchComboBox(this, w);
+        box->setIconSize(QSize(16, 16));
+        box->setToolTip(_action->toolTip());
+        box->setStatusTip(_action->statusTip());
+        box->setWhatsThis(_action->whatsThis());
+        box->addActions(_group->actions());
+        connect(_group, SIGNAL(triggered(QAction*)), box, SLOT(onActivated(QAction*)));

```

Comment:
Ok done.

Reply:
In new code the old-style connect() shouldn't be used any more. Use the new-style version instead. Because onActivated is a member function of WorkbenchComboBox the argument of the lambda expression must be adjusted.